core/test/matrix/dense.cpp

```diff
@@ -407,4 +407,65 @@ TYPED_TEST(Dense, CanCreateRealView)
 }
 
 
+TYPED_TEST(Dense, CanMakeMutableView)
+{
+    auto view = gko::make_dense_view(this->mtx.get());
+
+    ASSERT_EQ(view->get_values(), this->mtx->get_values());
+}
+
+
+TYPED_TEST(Dense, CanMakeConstView)
+{
+    auto view = gko::make_const_dense_view(this->mtx.get());
+
+    ASSERT_EQ(view->get_const_values(), this->mtx->get_const_values());
+}
+
+
+class CustomDense : public gko::EnableLinOp<CustomDense, gko::matrix::Dense<>> {
+    friend class gko::EnablePolymorphicObject<CustomDense,
+                                              gko::matrix::Dense<>>;
+
+public:
+    static std::unique_ptr<CustomDense> create(
+        std::shared_ptr<const gko::Executor> exec, gko::dim<2> size)
+    {
+        return std::unique_ptr<CustomDense>(
+            new CustomDense(std::move(exec), size));
+    }
+
+private:
+    explicit CustomDense(std::shared_ptr<const gko::Executor> exec,
+                         gko::dim<2> size = {})
+        : gko::EnableLinOp<CustomDense, gko::matrix::Dense<>>(std::move(exec),
+                                                              size)
+    {}
+};
+
+
+TEST(DenseView, MutableViewKeepsRuntimeType)
+{
+    auto vector = CustomDense::create(gko::ReferenceExecutor::create(),
+                                      gko::dim<2>{3, 4});
+
+    auto view = gko::make_dense_view(vector.get());
+
+    ASSERT_EQ(view->get_values(), vector->get_values());
+    ASSERT_TRUE(dynamic_cast<CustomDense*>(view.get()));
+}
+
+
+TEST(DenseView, ConstViewKeepsRuntimeType)
+{
+    auto vector = CustomDense::create(gko::ReferenceExecutor::create(),
+                                      gko::dim<2>{3, 4});
+
+    auto view = gko::make_const_dense_view(vector.get());
+
+    ASSERT_EQ(view->get_const_values(), vector->get_const_values());
+    ASSERT_TRUE(dynamic_cast<const CustomDense*>(view.get()));
```

Comment:
If CustomDense contains some additional data, does it keeps those data?
For example, CustomDense contains a int and set it to 5 (2 by default).
will the view use 5 or 2?

Reply:
The view will not keep the data. I don't think there is a way to let it keep the data, without overriding `create_view_of_impl` by the user. I will remove that behavior and always just return a `Dense`.